lib/storage/src/content_manager/toc.rs

```diff
@@ -356,14 +376,42 @@ impl TableOfContent {
         operation: CollectionMetaOperations,
         wait_timeout: Option<Duration>,
     ) -> Result<bool, StorageError> {
+        // if distributed deployment is enabled
         if self.propose_sender.is_some() {
+            let shard_distribution = match &operation {
+                CollectionMetaOperations::CreateCollection(op) => {
+                    let shard_number = op.create_collection.shard_number;
+                    let known_peers: Vec<_> = self
+                        .raft_state
+                        .lock()?
+                        .peer_address_by_id
+                        .read()?
+                        .keys()
+                        .cloned()
+                        .collect();
+                    let known_collections = self.collections.read().await;
+                    let known_shards: Vec<_> = known_collections
+                        .iter()
+                        .flat_map(|(_, col)| col.all_shards())
+                        .collect();
+                    let shard_distribution = ShardDistributionProposal::build(
+                        shard_number,
+                        self.this_peer_id,
+                        &known_peers,
+                        known_shards,
+                    );
+                    log::info!("Proposing distribution for {} shards for collection '{}' among {} peers {:?}", shard_number, op.collection_name, known_peers.len(), shard_distribution.distribution);
```

Comment:
Maybe implement a Display trait for shard distribution for more beautiful logs? Though I haven't checked - maybe debug is already good :)